src/test/scala/nl/knaw/dans/easy/transform/XmlTransformationSpec.scala

```diff
@@ -0,0 +1,104 @@
+    File(getClass.getResource("/metadata/").toURI).copyTo(metadataDir)
+  }
+
+  "enrichFilesXml" should "leave in the first file element accessibleToRights and visibleToRights elements as they were" in {
+    val filesXml = XML.loadFile(files_open)
+    val datasetXml = XML.loadFile(dataset_open)
+    val firstFileElement = (xmlTransformation.enrichFilesXml(filesXml, datasetXml, downloadUrl) \ "file").head
+    (firstFileElement \ "accessibleToRights").text shouldBe "NONE"
+    (firstFileElement \ "visibleToRights").text shouldBe "RESTRICTED_REQUEST"
+  }
+
+  it should "add a new <source> element in file elements" in {
+    val filesXml = XML.loadFile(files_open)
+    val datasetXml = XML.loadFile(dataset_open)
+    val origSizeFirstFileElement = (filesXml \ "file").head.child.size
+    val firstFileElement = (xmlTransformation.enrichFilesXml(filesXml, datasetXml, downloadUrl) \ "file").head
+    val sourceElement = (xmlTransformation.enrichFilesXml(filesXml, datasetXml, downloadUrl) \ "file" \ "source").head
+    sourceElement.child.size shouldBe 1
+    firstFileElement.child.size shouldBe origSizeFirstFileElement + 1
+  }
+
+  it should "give download path as value for the new <source> element in all file elements" in {
+    val filesXml = XML.loadFile(files_open)
+    val datasetXml = XML.loadFile(dataset_open)
+    val sourceElement_1 = (xmlTransformation.enrichFilesXml(filesXml, datasetXml, downloadUrl) \ "file" \ "source").head
+    val sourceElement_2 = (xmlTransformation.enrichFilesXml(filesXml, datasetXml, downloadUrl) \ "file" \ "source") (1)
+    sourceElement_1.text shouldBe "https://download/location/data/path/to/file.txt"
+    sourceElement_2.text shouldBe "https://download/location/data/quicksort.hs"
+  }
+
+  it should "add accessibleToRights element with value 'ANONYMOUS' to the second file element, when dataset accessRights is OPEN_ACCESS" in {
+    val filesXml = XML.loadFile(files_open)
+    val datasetXml = XML.loadFile(dataset_open)
+    val secondFileElement = (xmlTransformation.enrichFilesXml(filesXml, datasetXml, downloadUrl) \ "file") (1)
+    (secondFileElement \ "accessibleToRights").text shouldBe "ANONYMOUS"
+  }
+
+  it should "add accessibleToRights element with value 'RESTRICTED_REQUEST' to the second file element, when dataset accessRights is REQUEST_PERMISSION" in {
+    val filesXml = XML.loadFile(files_request)
+    val datasetXml = XML.loadFile(dataset_request)
+    val secondFileElement = (xmlTransformation.enrichFilesXml(filesXml, datasetXml, downloadUrl) \ "file") (1)
+    (secondFileElement \ "accessibleToRights").text shouldBe "RESTRICTED_REQUEST"
+  }
+
+  it should "add accessibleToRights element with value 'NONE' to the second file element, when dataset accessRights is NO_ACCESS" in {
+    val filesXml = XML.loadFile(files_no)
+    val datasetXml = XML.loadFile(dataset_no)
+    val secondFileElement = (xmlTransformation.enrichFilesXml(filesXml, datasetXml, downloadUrl) \ "file") (1)
```

Comment:
* also test that the first element didn't change in regards to the `accessibleToRights` and `visibileToRights`
* also test that the second element added a `visibleToRights`
* not sure, but I think I'm missing tests for the `<dcterms:source>` element

Apply this to the other tests as well.

Reply:
Done.  There were already tests for the `<dcterms:source>` element